src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb

```diff
@@ -5507,6 +5507,33 @@ class C
             End Using
         End Function
 
+        <WpfTheory, Trait(Traits.Feature, Traits.Features.Completion)>
+        <MemberData(NameOf(AllCompletionImplementations))>
+        <WorkItem(38297, "https://github.com/dotnet/roslyn/issues/38297")>
+        Public Async Function SelectionMustBeCaseSensitiveWithEnums(completionImplementation As CompletionImplementation) As Task
+            Using state = TestStateFactory.CreateCSharpTestState(completionImplementation,
+                <Document><![CDATA[
+class C
+{
+    void M(SampleEnum sampleEnum)
+    {
+        N($$
+    }
+
+    void N(SampleEnum sampleEnum) { }
+    enum SampleEnum
+    {
+        One,
+        Two
+    }
+}]]>
+                </Document>)
+
+                state.SendTypeChars("sam")
+                Await state.AssertSelectedCompletionItem("sampleEnum")
```

Comment:
The change in this particular scenario is fine, although I personally think the original behavior is OK too, since the user might want to type `SampleEnum.One` instead of the parameter. That said, I'm concerned about the potential behavior change in selection caused by not ignoring case anymore. There's comment explicitly explains why do we do comparison in case insensitive fashion, and this change would break that.

Reply:
Thank you, @genlu! Please note we keep the preference of capitalized enums for Ctrl + J.

Reply:
> Please note we keep the preference of capitalized enums for Ctrl + J.

Hmm... Why do we prefer a different behavior in ctrl+j sceanrio?

Reply:
Thank you, @genlu! Here are three scenarios:
1. Start typing with a Captial. We prefer a Capitalized symbol, e.g. an Enum name
2. Start typing with a small. We prefer a symbol started with a small letter, e.g. variable name.
3. Press CTRL+J just after dot, we prefer Enum because nothing is typed and with just go to the MatchPriority and the Enum Match priority was set intentionally higher than all others.